nemo_skills/pipeline/utils.py

```diff
@@ -46,6 +46,52 @@
 REUSE_CODE_EXP = {}
 
 
+@dataclass
+class RepoMetadata:
+    """Metadata for a repo that is used in the experiment."""
+
+    name: str
+    path: Path
```

Comment:
Im creating this dataclass because in the future, we may want to allow other repos to specifically add `include_patterns` and `include_patterns_relative_path` - for now, we just add the git level repo clone only (sufficient for my use case)